[17.0][MIG] purchase_order_secondary_unit: Migration to 17.0

mmrondon opened the pull request.

sergio-teruel committed:
purchase_order_secondary_unit: New module to allow buy in secondary units

Nikul-Chaudhary committed:
[MIG] purchase_order_secondary_unit v11 to v12

ps-tubtim committed:
[IMP] purchase_order_secondary_unit: black, isort

ps-tubtim committed:
[MIG] purchase_order_secondary_unit: Migration to 13.0

sergio-teruel committed:
[IMP] purchase_order_secondary_unit: more clear string field
TT27363

sergio-teruel committed:
[IMP] purchase_order_secondary_unit: Less code. Use product_secondary_unit mixin model from product-attribute repository.

kittiu committed:
[13.0][FIX] purchase_order_secondary_unit, error selecting 2nd_uom when no product_id

SurachetS committed:
[MIG] purchase_order_secondary_unit: Migration to 14.0

kakurai8 committed:
Added translation using Weblate (Japanese)

kh-badep committed:
Added translation using Weblate (French)

HviorForgeFlow committed:
[FIX] po_secondary_unit: Drop 0 qty test check on pol with secondary UoM

marylla committed:
Added translation using Weblate (German)

BT-mgomez committed:
[IMP] purchase_order_secondary_unit: black, isort, prettier

BT-mgomez committed:
[15.0][MIG] purchase_order_secondary_unit: Migration to 15.0

Commit:
[IMP] purchase_order_secondary_unit: Compatibility with product variants

HviorForgeFlow committed:
[FIX] purchase_order_secondary_unit: Split tests

sergio-teruel committed:
[IMP] purchase_order_secondary_unit: Add secondary units to purchase order reports

yajo committed:
[MIG] purchase_order_secondary_unit: Migration to v16

sergio-teruel committed:
[FIX] purchase_order_secondary_unit: User can changes secondary uom in confirmed purchase

pilarvargas-tecnativa committed:
[FIX] purchase_order_secondary_unit: Set product unit of measure for purchases

TT45090

acsonefho committed:
[FIX] purchase_order_secondary_unit: precompute for some fields & sec qty set to 1

Co-authored-by: David Alonso // Solvos <david.alonso@solvos.es>

carlosdauden committed:
[IMP] purchase_order_secondary_unit: Require secondary_uom_id when secondary_uom_qty is filled

dalonsod committed:
[FIX] purchase_order_secondary_unit: bug calling super()

Fix a bug when calling super's `_compute_product_qty()` was skipping the compute helper from the mixin.

@moduon MT-3483

Co-authored-by: Andrea Cattalani <22261939+anddago78@users.noreply.github.com>

pilarvargas-tecnativa committed:
[IMP] purchase_order_secondary_unit: make the secondary unit and secondary quantity fields optional

TT49686

mmrondon commented:
> Depending PR:
>
>  * [1560] [[17.0][MIG] product_secondary_unit product-attribute#1560](https://github.com/OCA/product-attribute/pull/1560)

alan196 reviewed (approved):
Functional Review 👍

sergio-teruel committed:
[FIX] purchase_order_secondary_unit: Error when user try to select a secondary uom without product in sale order line
TT51683

Commit:
[UPD] Update purchase_order_secondary_unit.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_secondary_unit/

dalonsod reviewed (changes requested):
@mmrondon at last, this addon has been merged for v16 (see #2216) and has some improvements that come from v15 and aren't still available here. Could you take a look?

cc @sergio-teruel

mymage committed:
Added translation using Weblate (Italian)

sergio-teruel commented:
Please @mmrondon Can yo sure that all commits from older version are in your PR..?
Thanks

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (11 of 11 strings)

Translation: purchase-workflow-16.0/purchase-workflow-16.0-purchase_order_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-16-0/purchase-workflow-16-0-purchase_order_secondary_unit/it/

dalonsod commented:
Hello @mmrondon will you resume this PR soon? Thanks!

peluko00 commented:
Hi @mmrondon, can you squash commits into one please?

mmrondon committed:
[MIG] purchase_order_secondary_unit: Migration to 17.0

[IMP] purchase_order_secondary_unit: pre-commit auto fixes

[IMP] purchase_order_secondary_unit: pre-commit auto fixes

 [DON'T MERGE] test-requirements.txt

[Don't MERGE] Remove dependency

[MIG] purchase_order_secondary_unit: Fix issue on test

[IMP] purchase_order_secondary_unit: pre-commit auto fixes
[IMP] purchase_order_secondary_unit: Restore commit history from version 16.0

dalonsod commented on purchase_order_secondary_unit/views/purchase_order_views.xml:
```diff
@@ -11,42 +11,42 @@
                     name="secondary_uom_qty"
-                    attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
                     groups="uom.group_uom"
-                    optional="show"
```
Please don't remove this, as it was added in an recent commit, when updating commit history

dalonsod commented on purchase_order_secondary_unit/views/purchase_order_views.xml:
```diff
@@ -11,42 +11,42 @@
                     options="{'no_create': True}"
-                    attrs="{'readonly': [('state', 'in', ('purchase','done', 'cancel'))], 'required': [('secondary_uom_qty', '!=', 0.0)]}"
                     groups="uom.group_uom"
-                    optional="show"
```
The same as above

dalonsod commented on purchase_order_secondary_unit/views/purchase_order_views.xml:
```diff
@@ -11,42 +11,42 @@
-                        attrs="{'readonly': [('state', 'in', ('purchase', 'done', 'cancel'))], 'required': [('secondary_uom_qty', '!=', 0.0)]}"
-                    />
-                </t>
+                <field
+                    name="secondary_uom_qty"
+                    class="oe_inline oe_no_button"
+                    groups="uom.group_uom"
+                    readonly="state in ['done', 'cancel']"
+                />
+                <field
+                    name="secondary_uom_id"
+                    class="oe_inline"
+                    domain="['|', ('product_id', '=', product_id),
+                                '&amp;', ('product_tmpl_id.product_variant_ids', 'in', [product_id]),
+                                         ('product_id', '=', False)]"
+                    options="{'no_create': True}"
+                    groups="uom.group_uom"
+                    readonly="state in ['purchase','done', 'cancel']"
+                    required="secondary_uom_qty != 0.0"
+                />
```
It seems that some later changes have been lost in this commit:
- Move whole code to a `<t>` node.
- Improvements in `secondary_uom_id` domain added by https://github.com/OCA/purchase-workflow/pull/2216/commits/ee8f610f2fed6f4456a5b4c8c131883e511170ac

dalonsod commented on purchase_order_secondary_unit/views/product_views.xml:
```diff
@@ -8,12 +8,11 @@
-                        options="{'no_create': True}"
-                    />
-                </t>
+                <field
+                    name="purchase_secondary_uom_id"
+                    options="{'no_create': True}"
+                    groups="uom.group_uom"
+                />
```
The same as below with `<t>` node

dalonsod reviewed (changes requested):
@mmrondon could take a look at the comments, and the changes you've added in migration commit in general? It seems that you've preserved original migration commit files changes after you've update commit history, and some changes added in recently added commits have been lost.

mmrondon commented:
hello @dalonsod sorry for that, already modified

dalonsod commented on purchase_order_secondary_unit/views/product_views.xml:
```diff
@@ -12,6 +12,7 @@
                     <field
                         name="purchase_secondary_uom_id"
                         options="{'no_create': True}"
+                        groups="uom.group_uom"
```
This is redundant, the field is placed into a `<t>` that have already groups attribute

dalonsod commented on purchase_order_secondary_unit/views/purchase_order_views.xml:
```diff
@@ -15,16 +15,19 @@
                         name="secondary_uom_qty"
                         class="oe_inline oe_no_button"
-                        attrs="{'readonly': [('state', 'in', ('done', 'cancel'))]}"
+                        groups="uom.group_uom"
```
The same as above

dalonsod commented on purchase_order_secondary_unit/views/purchase_order_views.xml:
```diff
@@ -15,16 +15,19 @@
+                                             ('product_id', '=', False)]"
                         options="{'no_create': True}"
-                        attrs="{'readonly': [('state', 'in', ('purchase', 'done', 'cancel'))], 'required': [('secondary_uom_qty', '!=', 0.0)]}"
+                        groups="uom.group_uom"
```
The same as above

dalonsod commented on purchase_order_secondary_unit/views/purchase_order_views.xml:
```diff
@@ -15,16 +15,19 @@
                         name="secondary_uom_id"
                         class="oe_inline"
-                        domain="product_id and ['|', ('product_id', '=', product_id),
+                        domain="['|', ('product_id', '=', product_id),
```
`domain` definition is still the old one, changes added by https://github.com/OCA/purchase-workflow/commit/ee8f610f2fed6f4456a5b4c8c131883e511170ac are lost, please check diff at this point.

dalonsod reviewed (changes requested):
There are still some things to be fixed in commit history, please check

dalonsod commented:
Hello @mmrondon would you resume this PR and attend pending comments? Thanks!

dalonsod commented:
Superseded by #2492
